### PR TITLE
MDSMonitor: add command to bring cluster down rapidly

### PR DESCRIPTION
**Pull request by batrick**
http://tracker.ceph.com/issues/37085

### CODE REVIEW

File: src/mon/FSCommands.cc [hunk 1]
```diff
@@ -76,6 +76,61 @@ class FlagSetHandler : public FileSystemCommandHandler
...
+      ss << "Not found: '" << fs_name << "'";
+      return -ENOENT;
+    }
+
+    auto f = [](auto fs) {
+      fs->mds_map.set_flag(CEPH_MDSMAP_NOT_JOINABLE);
+    };
+    fsmap.modify_filesystem(fs->fscid, std::move(f));
+
+    std::vector<mds_gid_t> to_fail;
```
Comment (fullerdj) [inline src/mon/FSCommands.cc]: Why is this needed? Can't you just use a single loop here?

Comment (batrick) [reply src/mon/FSCommands.cc]: I don't think we can safely iterate through a map when we're deleting entries (via `MDSMonitor::fail_mds_gid`). See also `MDSMap::get_mds_info`.

**Comment by batrick** [comment]
retest this please
